File: no 1/java/memory.java

// Latihan 1

class Memory extends Hardware{
	private String frequency;
	private String memorySize;
	private String supportsCuda;

	// constructor
	Memory(){
	}

	public void setFrequency(String frequency){
		this.frequency = frequency;
	}
	public String getFrequency(){
		return this.frequency;
	}
	
	public void setMemorySize(String memorySize){
		this.memorySize = memorySize;
	}
	public String getMemorySize(){
		return this.memorySize;
	}

	public void setSupportsCuda(String supportsCuda){
		this.supportsCuda = supportsCuda;
	}
	public String getSupportsCuda(){
		return this.supportsCuda;
	}
}

File: no 2/python/memory.py
class Memory:

# kelas yang digunakan untuk mengimplementasikan kelas memory

	def __init__(self):
	# konstruktor
		self.frequency = ""
		self.memorySize = ""
		self.supportsCuda = ""

	def setFrequency(self, frequency):
	# set nilai atribut frequency
		self.frequency = frequency;

	def getFrequency(self):
	# mengembalikan nilai atribut frequency
		return self.frequency

	def setMemorySize(self, memorySize):
	# set nilai atribut memorySize
		self.memorySize = memorySize
	def getMemorySize(self):
	# mengembalikan nilai atribut memorySize
		return self.memorySize

	def setSupportsCuda(self, supportsCuda):
	# set nilai atribut supportsCuda
		self.supportsCuda = supportsCuda
	def getSupportsCuda(self):
	# mengembalikan nilai atribut supportsCuda
		return self.supportsCuda